lib/constructs/interface.js

```diff
@@ -627,8 +635,18 @@ class Interface {
         `;
       } else if (this.iterable.isPair) {
         this.str += `
-          exports.createDefaultIterator = (target, kind) => {
-            const iterator = Object.create(IteratorPrototype);
+          exports.createDefaultIterator = (globalObject, target, kind) => {
+            const ctorRegistry = globalObject[ctorRegistrySymbol];
+            if (ctorRegistry === undefined) {
+              throw new Error('Internal error: invalid global object');
+            }
+
+            if (ctorRegistry["${this.name}"] === undefined) {
+              throw new Error('Internal error: constructor ${this.name} is not installed on the passed global object');
+            }
```

Comment:
I guess we should add these to stay consistent, but I am regretting them now; we should just remove all these checks and let runtime errors get thrown by attempting to access properties of undefined. They are not worth the generated code size.

Let's make sure we get them removed before the next release.